src/ui/flutter_app/test/helpers/test_mills.dart

```diff
@@ -1,25 +1,98 @@
-const testMill = " 1.    g1    b4"
-    "  2.    f2    f4"
-    "  3.    d1    d6"
-    "  4.    a1xf4    f4"
-    "  5.    g4    d2"
-    "  6.    g7xf4    f4"
-    "  7.    c3    c5"
-    "  8.    b6    a7"
-    "  9.    d3    e3"
-    " 10.    c3-c4    d2-b2"
-    " 11.    d1-d2    e3-e4"
-    " 12.    d2-d1xb2    c5-d5"
-    " 13.    f2-d2xd6    b4-a4"
-    " 14.    c4-b4    f4-f2"
-    " 15.    g7-d7    e4-f4"
-    " 16.    d3-e3    d5-d6"
-    " 17.    e3-e4    d6-f6xb4"
-    " 18.    d2-b2    f2-d2"
-    " 19.    d7-g7xf4    a7-d7"
-    " 20.    g4-f4    a4-b4"
-    " 21.    f4-g4xf6    d2-d5"
-    " 22.    d1-d2    b4-d6xd2"
-    " 23.    g4-f4    d5-g4"
-    " 24.    b2-d2    g4-d5xd2"
-    " 25.    g1-g4xd6";
+/*
+  This file is part of Sanmill.
+  Copyright (C) 2019-2021 The Sanmill developers (see AUTHORS file)
+
+  Sanmill is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Sanmill is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+abstract class TestGame {
+  const TestGame();
+
+  String get export;
+
+  String? get movesSinceRemove;
+
+  String get recorder;
+
+  String get exportNotStandard;
+}
+
+class WinnGame extends TestGame {
```

Comment:
My thinking was that we'll have a few scenarios to test. For example the export you see in there was where the player one has won the game because the other player had less than three pieces.

Therefore we can't test the `movesSinceLastRemove` (as the last remove was the last move) nor can we test other GameOverReasons. The idea is that we have a few classes with different games and iterate over them to verify multiple cases.

I named it WinnGame as it has a recorded game (the export) and a few other metadata about it to verify our code.

How would you name it? Maybe something like WinnLessThanThreeGame ???

Reply:
Maybe it is OK. Win, not Winn.